pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/AttributeAxisIterator.java

```diff
@@ -35,7 +35,8 @@
 
     /* Constants used to determine which methods are accessors */
     private static final Set<Class<?>> CONSIDERED_RETURN_TYPES
-            = new HashSet<>(Arrays.<Class<?>>asList(Integer.TYPE, Boolean.TYPE, Double.TYPE, String.class, Long.TYPE, Character.TYPE, Float.TYPE));
+            = new HashSet<>(Arrays.<Class<?>>asList(Integer.TYPE, Boolean.TYPE, Double.TYPE, String.class,
+                    Long.TYPE, Character.TYPE, Float.TYPE, List.class));
```

Comment:
I think we should be a little more careful here. Many nodes have getters which produce `List` instances whose component type is not currently supported as an attribute value (eg `List<Node>`). We should either query the getter's `Method::getGenericReturnType` and assert the type parameter on `List` is ok, or use an annotation to opt-in the getter when it's a list. 

I fear, that way too many attributes may become available, when they weren't intended as such. We should have some kind of test to check that.

Reply:
Also, isn't `Enum.class` missing from this list?

Reply:
I've restricted it now to the types, we already allow. We'll have with PMD 7 the `@NoAttribute` annotation which helps in removing any method that should not be exposed.